lib/daru/maths/statistics/vector.rb

```diff
@@ -30,7 +30,9 @@ def median
         end
 
         def mode
-          frequencies.max { |a,b| a[1]<=>b[1] }.first
+          max = frequencies.values.max
+          output = frequencies.select { |_k,v| v == max }.to_a.map(&:first)
```

Comment:
No need to call `to_a.map`. Just `.values` will suffice since `select` will return a `Hash`.